### PR TITLE
Add failed output verification after task was mined

### PR DESCRIPTION
**Pull request by lenak25**

### CODE REVIEW

File: src/worker/controller/actions/sync/ReceiveAllPipelineAction.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 const constants = require("../../../../common/constants");
 const errs = require("../../../../common/errors");
 const NODE_NOTIFY = constants.NODE_NOTIFICATIONS;
-const waterfall = require("async/waterfall");
```
Comment (assafmo) [inline src/worker/controller/actions/sync/ReceiveAllPipelineAction.js:4]: :tada:

File: test/verifier_test.js [hunk 2]
```diff
@@ -1979,4 +1985,27 @@ describe("Verifier tests", function() {
...
+  it("Wrong output hash of failed task submission pre-mined", async function() {
+    const tree = TEST_TREE.verifier;
+    if (!tree["all"] || !tree["#48"]) {
+      this.skip();
+    }
+
+    return new Promise(async function(resolve) {
+      let a = await initStuffForTaskSubmission();
+      let task = new FailedResult(a.taskId, constants.TASK_STATUS.FAILED, a.output, 5, "signature");
+      let status = constants.ETHEREUM_TASK_STATUS.RECORD_CREATED;
```
Comment (assafmo) [inline test/verifier_test.js]: Please use `const`

File: test/verifier_test.js [hunk 3]
```diff
@@ -1979,4 +1985,27 @@ describe("Verifier tests", function() {
...
+
+      a.apiMock.setTaskParams(a.taskId, a.blockNumber, status);
+      a.verifier.verifyTaskSubmission(task, a.blockNumber, a.taskId, null).then(res => {
+        assert.strictEqual(res.isVerified, false);
+        assert.strictEqual(res.error instanceof errors.TaskVerificationErr, true);
+        resolve();
```
Comment (assafmo) [inline test/verifier_test.js:2004]: I think that if a test returns a Promise you don't need to resolve it.

**Review by assafmo — DISMISSED** [review]

File: test/verifier_test.js (hunk 3 above)
Comment (lenak25) [reply test/verifier_test.js:2004]: It is required

**Review by assafmo — APPROVED** [review]

**Pull request merged**